lib/rules/no-duplicate-imports.js

```diff
@@ -21,18 +57,62 @@ function getValue(node) {
     return "";
 }
 
+
+/**
+ * Returns a boolean if we should report contradictory import type.
+ * @param {string[]} importTypes An array contain import types of a module.
+ * @param {string} importType An contradictory import type to check if we should report it or not.
+ * @returns {boolean} true if the contradictory import type should be reported.
+ */
+function shouldReportContradictoryImportType(importTypes, importType) {
+    if (importTypes.indexOf(importType) > -1) {
+        return true;
+    }
+    return (
+        importTypes.findIndex(importTypeItem => CONTRADICTORY_IMPORT_TYPES.includes(importTypeItem)) === -1
+    );
+
+}
+
 /**
  * Checks if the name of the import or export exists in the given array, and reports if so.
  * @param {RuleContext} context The ESLint rule context object.
  * @param {ASTNode} node A node to get.
  * @param {string} value The name of the imported or exported module.
  * @param {string[]} array The array containing other imports or exports in the file.
  * @param {string} messageId A messageId to be reported after the name of the module
+ * @param {{}} modulesWithImportTypes The object containing the name of unique modules with their first import type [specificImport, nameSpaceImport].
+ * @param {string} type the name of import type.
+ * @param {string[]} ExportAllDeclarationsInFile The array containing ExportAllDeclarations in the file.
  *
  * @returns {void} No return value
  */
-function checkAndReport(context, node, value, array, messageId) {
-    if (array.indexOf(value) !== -1) {
+function checkAndReport(
+    context,
+    node,
+    value,
+    array,
+    messageId,
+    modulesWithImportTypes,
+    type,
+    ExportAllDeclarationsInFile
+) {
```

Comment:
This looks like too many parameters. Can we pass just `context`, `node`, `messageId`, and previous declarations?

Reply:
Yes, we can use  4 parameters instead of 8, the same thing for the functions `handleExports `and `handleImports`.